Sprint 4/step 1

sadova-nadezhda opened the pull request:
Проект. Типизация

sadova-nadezhda committed:
update ts

utils updated
types updated
services updated

sadova-nadezhda committed:
update ts

updated services auth

sadova-nadezhda committed:
update ts

сломался автологин

sadova-nadezhda committed:
update ts

kyzinatra commented on src/components/App/App.tsx:
```diff
@@ -8,7 +8,6 @@ import {
-  ResetPasswordPage,
   NotFoundPage,
 } from "../../pages";
 import ProfileForm from "../Form/ProfileForm";
```
Надо исправить: ProtectedRoute также должен защищать авторизованных пользователей от того, чтобы попасть на страницы вроде `/login`, `/register` итд

kyzinatra commented on src/components/App/App.tsx:
```diff
@@ -8,7 +8,6 @@ import {
   ForgotPasswordPage,
-  ResetPasswordPage,
   NotFoundPage,
 } from "../../pages";
```
Отлично! Отличная реализация модальных окон

kyzinatra commented on src/components/Form/LoginForm.tsx:
```diff
@@ -14,7 +14,7 @@ export default function LoginForm() {
 
-  const handleSubmit = async (e: FormEvent) => {
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
```
Отлично! Хорошая типизация

kyzinatra commented on src/hooks/useForm.ts:
```diff
@@ -1,14 +1,14 @@
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>) => {
     const { name, value } = e.target;
     setValues((prevValues) => ({ ...prevValues, [name]: value }));
```
Отлично! Понравилось, что вы используете `useForm` хук

kyzinatra commented on src/utils/checkResponse.ts:
```diff
@@ -1,3 +1,11 @@
+      return Promise.reject(new Error('Ошибка парсинга JSON'));
+    }
+  }
+  const errorMessage = `Ошибка ${res.status}: ${res.statusText}`;
```
Надо исправить: в случае получения ошибки `token expired` нужно обновить токен и повторить запрос

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. Но все же есть моменты на которые стоит обратить внимание.
Подробнее: 

**Что важно исправить:**

-  ProtectedRoute также должен защищать авторизованных пользователей от того, чтобы попасть на страницы вроде `/login`, `/register` итд
-  в случае получения ошибки `token expired` нужно обновить токен и повторить запрос


**Что понравилось:**

-  Отличная реализация модальных окон
-  Хорошая типизация
-  Понравилось, что вы используете `useForm` хук


**Обратите внимание, что работа принимается только после исправления всех `«Надо исправить»`.**

sadova-nadezhda committed:
update ProtectedRoute

sadova-nadezhda committed:
update project

- обновила логику автологина (после изменений данных пользователя при обновлении не срабатывал автологин)
- обновила ProtectedRoute
- в случае получения ошибки token expired обновляем токен и повторяем запрос
- настроила маршруты

sadova-nadezhda committed:
Update ProfilePage.tsx

kyzinatra commented:
Вы отлично потрудились.
Поздравляю! Ваша работа принята.
Удачного дальнейшего обучения.
